deploy-manage/upgrade/deployment-or-cluster/upgrade-717.md

```diff
@@ -1,7 +1,501 @@
+
+For additional details and best practices, review the [{{es}} upgrade setup guide](https://www.elastic.co/guide/en/elasticsearch/reference/8.19/setup-upgrade.html).
+
+As part of your preparation, make sure to complete all tasks reported by the **Upgrade Assistant**, review any installed plugins for compatibility, and check whether custom client applications are affected by API-related breaking changes so you can address them before the upgrade.
+::::
+
+:::::
+
+### 8.19 upgrade execution
+
+Keep the following considerations in mind when upgrading your deployment or cluster:
+
+* If you use [Stack monitoring](/deploy-manage/monitor/stack-monitoring.md) with a dedicated monitoring cluster, upgrade your monitoring cluster first.
+* If you use [remote cluster](/deploy-manage/remote-clusters.md) functionality, upgrade the remote clusters first.
+* Before starting the upgrade, run the same checks and validations you plan to perform afterward, so you have a baseline for comparison. Refer to the [upgrade validation](#819-validation) section for example checks.
+
+The steps below describe how to upgrade the core components of your {{stack}} environment, {{es}}, {{kib}}, and, when applicable, {{fleet-server}} and Elastic APM, for each deployment type.
+
+:::::{applies-switch}
+
+::::{applies-item} ess:
+
+To upgrade your deployment to 8.19, follow the steps in [Upgrade on Elastic Cloud → Perform the upgrade](https://www.elastic.co/guide/en/elastic-stack/8.19/upgrade-elastic-stack-for-elastic-cloud.html#perform-cloud-upgrade).
+
+During the upgrade process, all components of your deployment are upgraded in the expected order:
+- {{es}}
+- {{kib}}
+- Integrations Server ({{fleet-server}} and APM)
+- Enterprise Search
+::::
+
+::::{applies-item} ece:
+
+To upgrade your deployment to 8.19, follow the steps in [Upgrade on Elastic Cloud → Perform the upgrade](https://www.elastic.co/guide/en/elastic-stack/8.19/upgrade-elastic-stack-for-elastic-cloud.html#perform-cloud-upgrade). 
+
+:::{note}
+Although this guide refers to {{ecloud}}, the same steps apply to ECE deployments.
+:::
+
+During the upgrade process, all components of your deployment are upgraded in the expected order:
+- {{es}}
+- {{kib}}
+- Integrations Server ({{fleet-server}} and APM)
+- Enterprise Search
+::::
+
+
+::::{applies-item} eck:
+In ECK, upgrades are performed declaratively by updating the `spec.version` field in the resource manifest. Once the new version is applied, the operator automatically orchestrates the rolling upgrade, ensuring that each component is upgraded safely and in the correct order.
+
+To upgrade your cluster to 8.19, follow the steps in [Upgrade on ECK](/deploy-manage/upgrade/deployment-or-cluster/upgrade-on-eck.md), and start upgrading the {{es}} and {{kib}} resources that represent the cluster.
```

Comment:
annoying: can we say "In each manifest, update the `spec.version` field to `8.19`" explicitly here?

Reply:
done! looks much better now, thanks!